src/Symfony/Component/AssetMapper/Command/ImportMapInstallCommand.php

```diff
@@ -42,18 +45,26 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $finishedCount = 0;
         $progressBar = new ProgressBar($output);
         $progressBar->setFormat('<info>%current%/%max%</info> %bar% %url%');
-        $downloadedPackages = $this->packageDownloader->downloadPackages(function (string $package, string $event, ResponseInterface $response, int $totalPackages) use (&$finishedCount, $progressBar) {
-            $progressBar->setMessage($response->getInfo('url'), 'url');
-            if (0 === $progressBar->getMaxSteps()) {
-                $progressBar->setMaxSteps($totalPackages);
-                $progressBar->start();
-            }
 
-            if ('finished' === $event) {
-                ++$finishedCount;
-                $progressBar->advance();
-            }
-        });
+        try {
+            $downloadedPackages = $this->packageDownloader->downloadPackages(function (string $package, string $event, ResponseInterface $response, int $totalPackages) use (&$finishedCount, $progressBar) {
+                $progressBar->setMessage($response->getInfo('url'), 'url');
+                if (0 === $progressBar->getMaxSteps()) {
+                    $progressBar->setMaxSteps($totalPackages);
+                    $progressBar->start();
+                }
+
+                if ('finished' === $event) {
+                    ++$finishedCount;
+                    $progressBar->advance();
+                }
+            });
+        } catch (LogicException|TimeoutException|TransportException $throwable) {
```

Comment:
The command should not be aware of any HTTP client exception. That's an implementation detail of the `RemotePackageDownloader`. A proper solution would be to catch the HTTP client exception in the downloader and upcast them to the exception that we catch here.

Reply:
Thanks for your response! Since I am not that will known within the Symfony components could you confirm that catching the HTTP client exceptions should happen here?

https://github.com/symfony/symfony/blob/61387107e6585068f1c1d403d8502d3e906298e6/src/Symfony/Component/AssetMapper/ImportMap/RemotePackageDownloader.php#L73

Reply:
I've added it on the line above, and tried it again which seem to give the exact response as I need.